tests/test_anvl.py

```diff
@@ -0,0 +1,138 @@
+import pytest
+
+from codalib import anvl
+
+
+class Test_readANVLString(object):
+    def test_with_empty_string(self):
+        """
+        Check that readANVLString can handle an empty string.
+        """
+        actual = anvl.readANVLString('')
+        expected = {}
+        assert actual == expected
+
+    def test_with_simple_anvl_string(self):
+        """
+        Check that readANVLString parses single key value.
+        """
+        actual = anvl.readANVLString('key:value')
+        expected = {'key': 'value'}
+        assert actual == expected
+
+    def test_strips_left_whitespace_of_value(self):
+        """
+        Check that readANVLString only strips whitespace on the left side
+        of the value.
+        """
+        actual = anvl.readANVLString('key: value ')
+        expected = {'key': 'value '}
+        assert actual == expected
+
+    @pytest.mark.xfail
+    def test_string_without_colon(self):
+        """
+        IndexError is not an appropriate response and
+        will be changed in the future.  This test verifies the functions
+        behavior, but it has been marked as an expected failure because
+        it is not the desired functionality.
+        """
+        with pytest.raises(IndexError):
+            anvl.readANVLString('foo bar baz qux')
+
+    def test_with_comment(self):
+        """
+        Verify readANVLString can handle a comment with the `#` char
+        in the first column of a line.
+        """
+        anvl_string = (
+            """# this is comment\n
+            key1: value1\n
+            key2: value2\n
+            """
+        )
+
+        actual = anvl.readANVLString(anvl_string)
+        expected = {'key1': 'value1', 'key2': 'value2'}
+        assert actual == expected
+
+    @pytest.mark.xfail
+    def test_with_comment_char_not_in_first_column(self):
+        """
+        readANVLString should be able to pickup a comment char even
+        if whitespace precedes it.  This test verifies the functions
+        behavior, but it has been marked as an expected failure because
+        it is not the desired functionality.
+        """
+        anvl_string = '\t# this is comment\n'
```

Comment:
I don't think that is necessary. The function is looking for a `#` char at the 0 index of the line. The ANVL spec is unclear, but it seems to me that a `#` at any index should indicate a comment. I included this test mostly to document the current behavior. Also note that this is marked as a expected failure (and it currently is failing).